server/routerlicious/packages/lambdas/src/nexus/protocol.ts

```diff
@@ -0,0 +1,33 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { NetworkError } from "@fluidframework/server-services-client";
+import * as semver from "semver";
+
+export const ProtocolVersions = ["^0.4.0", "^0.3.0", "^0.2.0", "^0.1.0"];
+
+function selectProtocolVersion(connectVersions: string[]): string | undefined {
+	for (const connectVersion of connectVersions) {
+		for (const protocolVersion of ProtocolVersions) {
+			if (semver.intersects(protocolVersion, connectVersion)) {
+				return protocolVersion;
+			}
+		}
+	}
+	return undefined;
+}
+
+export function checkVersion(versions: string[]): [string[], string] {
```

Comment:
nit: checkProtocolVersion,

 thinking of someone reading the connect.ts and immediately know what this function does